devices/tplink_re200-v4.yml

```diff
@@ -0,0 +1,23 @@
+vendor: TP-Link
+model: RE200
+variant: v4
+bootloader: unknown
+cpu: MediaTek MT7628AN
+cpu_cores: 1
+cpu_mhz: 580
+memory_mb: 64
+storage:
+  - type: unknown
+    size_mb: 8
+device_type: WiFi Router
+ethernet_100m_ports: 1
+jtag: false
+serial: 57600 / 8N1
+outdoor: false
```

Comment:
As OpenWrt, I don't think this is really so much important to us; I would actually drop that parameter.

Reply:
This was requested by Thomas, in fact there are more "optional" values which seem valuable metadata. Just as the power supply or e.g. device dimensions. I see this data source not just as a base for developers but also for tooling around it which allows users to search for specific needs.

Reply:
But keep in mind that we are a router operating system/platform, and not a device feature catalogue.

I'd strongly recommend to limit this only to features somehow connected to the firmware or operation, and neither outdoor nor dimension are one of them.

Reply:
True, I don't mean to add subjective information like `very_very_fast: False` but store static facts that are valuable to both developers and users. I'd see device specifications as documentation and therefore wouldn't be to stingy on adding it, as long as it's consistent and static.

Adding `outdoor` as an optional parameter, or actually better [ip_code](https://en.wikipedia.org/wiki/IP_Code), doesn't seem to harmful for me.

Reply:
Having an optional parameter might be a suitable compromise here. I just hesitate to make this too open, as the main openwrt repo should not become a Wiki database eventually. With optional parameters, we quickly might end up with a lot of commits just changing meta data, e.g. stuff like "add dimensions for x", "add outdoor for Y", "add case color for Z".

If we open this for optional parameters, people will want to add/move more information here, and this might get ouf of hands or by just annoying quickly. So, I'm not against adding this additional information, I mostly fear the resulting problems this will probably create.

Reply:
Let's make dimension, power supply and ip_code non optional then?

Reply:
What's ip_code and power supply?

Reply:
ip_code: https://en.wikipedia.org/wiki/IP_Code
power_supply: https://en.wikipedia.org/wiki/Power_supply

I understand that these information are not directly of concern when developing software, however so isn't the memory size when device-tree is set to auto-detection.

Having metadata maintained (additionally) by developers should improve the collaboration between wiki and developers, which are at this point still fairly separated.

Reply:
> power_supply: https://en.wikipedia.org/wiki/Power_supply

Honestly?

Reply:
No this was clearly a joke, but you asked me, didn't you?

Reply:
The example section of `device.yaml` shows you how it's used:
https://github.com/openwrt/openwrt/pull/3534/files#diff-f7ab3dd4354a2c48efa41a6cb66edd8cb3469921a9233e06ae20c9984ecd0632R37

Reply:
> Having an optional parameter might be a suitable compromise here. I just hesitate to make this too open, as the main openwrt repo should not become a Wiki database eventually.

I strongly suggest to make this as open as possible, and to make this information as usefull for the users as possible. The information "outdoor yes/no" is easy to get, and highly usefull for users who are in need of an outdoor device.

Reply:
Rephrasing my goal, unite efforts of developers and wiki (aka documentation) maintainers. Maybe this should be part of `openwrt.git` but of `devices.git` and then maintained by a community (e.g. device owners) but required for adding devices to `openwrt.git`.

Workflow would be:
* generate device specific YAML file
* propose it to `devices.git` or send along a PR if added via mailing list
* commit message for `openwrt.git` contains the commit ID of the `devices.git` commit that initially adds the file

This way we can get unified device information prepared by developers and then (carefully) extended by the community.

However I see a break in the current data structures. Storing latest release in the metadata means frequent updates and redundancy to the automatically created `profiles.json` files. Also all added information should be validated via schema definitions, no random and inconsistent data should be possible.

Reply:
> My understanding of this PR was that this will eventually become the successor of the current dataentries / ToH. If that is not wanted (too much info unrelated to OpenWrt, too many commits for changing metadata), then we should immediately stop here and search for a different approach.

This should be discussed by more developers eventually; there might be very different opinions on whether the OpenWrt "main" repo should contain and track documentation like this. However, we can of course sharpen a few edges beforehand. Be aware that I was a fan of that idea (in general) from the start, there might be others entirely disliking it.

(Will comment on the details later.)

Reply:
> Be aware that I was a fan of that idea (in general) from the start, there might be others entirely disliking it.

Yes and I'm very glad that's the case! Once this is in better shape I'll propose it to the mailing list.